### ISSUE TITLE
pdfcpu val -v crashes on pdf that displays fine in chrome but crashes...

### ISSUE THREAD

**Issue by joel-rieke** [pdfcpu/pdfcpu#780]
Running the following:
pdfcpu val -v <redacted_pdf>

 INFO: 2024/01/18 14:20:05 PDF Version 1.5 conforming reader
 INFO: 2024/01/18 14:20:05 validating
STATS: 2024/01/18 14:20:05 XRefTable:
*************************************************************************************************
HeaderVersion: 1.6
has 1 pages
using object streams
using xref streams
is linearized file
XRefTable:
                     Size: 399
              Root object: (296 0 R)
              Info object: (294 0 R)

Stack trace ensues...  with complaint about types.Integer -- which appears to be a standard type and maybe should be supported?

Fatal: pdfcpu: validateNameEntry: dict=StructElementDict entry=S invalid type types.Integer
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateNameEntry
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/validate/object.go:691
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateStructElementDictPart1
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/validate/structTree.go:354
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateStructElementDict
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/validate/structTree.go:459
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateStructTreeRootDictEntryKArray
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/validate/structTree.go:487
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateStructTreeRootDictEntryK
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/validate/structTree.go:533
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateStructTreeRootDict
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/validate/structTree.go:621
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateStructTree
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/validate/structTree.go:690
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.validateRootObject
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/validate/xReftable.go:921
github.com/pdfcpu/pdfcpu/pkg/pdfcpu/validate.XRefTable
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/validate/xReftable.go:44
github.com/pdfcpu/pdfcpu/pkg/api.ValidateContext
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/api/api.go:80
github.com/pdfcpu/pdfcpu/pkg/api.Validate
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/api/validate.go:55
github.com/pdfcpu/pdfcpu/pkg/api.ValidateFile
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/api/validate.go:99
github.com/pdfcpu/pdfcpu/pkg/api.ValidateFiles
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/api/validate.go:122
github.com/pdfcpu/pdfcpu/pkg/cli.Validate
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/cli/cli.go:32
github.com/pdfcpu/pdfcpu/pkg/cli.Process
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/cli/process.go:35
main.process
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/cmd/pdfcpu/process.go:149
main.processValidateCommand
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/cmd/pdfcpu/process.go:206
main.commandMap.process
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/cmd/pdfcpu/cmd.go:143
main.main
	<redacted>/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/cmd/pdfcpu/main.go:56

* Your issue is based on the latest commit
Running @latest 0.6.0 release.

* State your OS and OS version (Debian bullseye)
Linux <redacted> 5.15.90.2-microsoft-standard-WSL2 #1 SMP Fri Jun 2 00:45:15 UTC 2023 x86_64 GNU/Linux

The pdf may be found here:
https://kingcounty.gov/~/media/depts/records-licensing/recorders-office/documents/MLAPPpdf.ashx?la=en

Any ideas on how to fix would be appreciated.

**Comment by joel-rieke** [comment]
Hm... does this work around the problem?  https://github.com/pdfcpu/pdfcpu/pull/762

**Comment by joel-rieke** [comment]
I was able to tweak/hack #762  to get this issue resolved.   Not sure if that would be acceptable or not however.

**Comment by joel-rieke** [comment]
> Your reported behaviour suggest a PDF spec violation. The S entry of the structure element dictionary is supposed to be a Name - in your case it's an Integer:
> 
> <img alt="Screenshot 2024-01-22 at 01 41 30" width="968" src="https://private-user-images.githubusercontent.com/11322155/298426917-67c2629c-9fff-4797-8b60-a2be2241a137.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MzgzMDQsIm5iZiI6MTcwNTkzODAwNCwicGF0aCI6Ii8xMTMyMjE1NS8yOTg0MjY5MTctNjdjMjYyOWMtOWZmZi00Nzk3LThiNjAtYTJiZTIyNDFhMTM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIyVDE1NDAwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjY2RiNDU4YWFhMzVjZjQxODI1MmZmODkwNDdhN2E4OTlkY2UwYzUzZjZkY2YxODVlYjA4YzA4NDU1MDJlZmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7e58Vhc7LsBWFE6ATGDp0DdnSW_GqGZfoTYRiH19yas">
> Normally I would consider extending relaxed validation, in this case I am not sure if that would make any sense.

I agree that leaving the normal validation alone is probably best.  However, somehow these pdf's available on public websites are breaking the specification.  Wondering if it could also be a parser issue however where the parser is seeing something that looks like an integer and is thus 'declaring it an integer' when it is just a string that happens to have a number (totally valid).

**Comment by hhrutter** [comment]
Try looking at the output of `pdfcpu validate -vv test.pdf`
This should output the xreftabl inspite of any parser error ideally.
Then we can verify your claim, or just fire up a hex editor.

I am happy to take care of this if you can share the file although I am pretty sure the content of S is missing a leading / which is the defined prefix for Name objects and therefore if gets interpreted as an Integer object.

**Comment by joel-rieke** [comment]
I put a link to the pdf in the description of the bug.  It's public url so, should be downloadable anywhere I believe.  It's just a marriage license pdf document.
